docs/admin/multiple-zones.md

```diff
@@ -0,0 +1,280 @@
+---
+---
+
+## Introduction
+
+Kubernetes 1.2 adds support for running a single cluster in multiple zones.
+This is a lightweight version of a broader effort for federating multiple
+Kubernetes clusters together (sometimes referred to by the affectionate
+nickname "Ubernetes").  Full federation will allow combining separate
+Kubernetes clusters running in different regions or clouds.  However, many
+users simply want to run a more available Kubernetes cluster in multiple zones
+of their cloud provider, and this is what the multizone support in 1.2 allows
+(we nickname this "Ubernetes Lite").
+
+Multizone support is deliberately limited: a single Kubernetes cluster can run
+in multiple zones, but only within the same region (and cloud provider).  Only
+GCE and AWS are currently supported automatically (though it would be easy to
+add similar support for other clouds or even bare metal, by simply arranging
+for the appropriate labels to be added to nodes and volumes).
+
+
+* TOC
+{:toc}
+
+## Functionality
+
+When nodes are started, the kubelet will automatically add labels to them with
+zone information.
+
+Just as pods in a replication controller or service are spread across nodes,
+the scheduler will use the zone labels to spread first across nodes in
+different zones (via `SelectorSpreadPriority`).  This is a best-effort
+placement, and so if one zone has less available capacity you should not be
+surprised if your pods are not scheduled evenly between the zones.
+
+When persistent volumes are created, labels are automatically added to them (by
+the `PersistentVolumeLabel` admission controller).  The scheduler (via the
+`VolumeZonePredicate` predicate) will then ensure that pods that claim that
+volume are only attached to volumes in their corresponding zone.
+ 
+## Limitations
+
+There are some important limitations of the multizone support.
+
+We assume that the different zones are located close to each other in the
+network, so we don't perform any zone-aware routing.  In particular, traffic
+that goes via services is likely to cross zones (even if pods in that service
+are in the same zone), and this may incur additional latency and cost.
+
+Volume zone-affinity will only work with a `PersistentVolume`, and will not
+work if you directly specify an EBS volume in the pod spec (for example).
+
+Clusters cannot span clouds or regions (this functionality will require full
+federation support).
+
+
+## Walkthough
+
+We're now going to walkthrough setting up and using a multi-zone cluster on both GCE & AWS.  The initial setup
+is still a little awkward, in that you bring up a full cluster (specifying `MULTIZONE=1`),
```

Comment:
OK!  I hope we can still make this better with some of mike danese's work, but removing the pre-emptory apology!